test/helpers/user.ts

```diff
@@ -286,3 +295,52 @@ export const createNewWalletFromPhone = async ({
 
   return wallet
 }
+
+export const createAndFundNewWalletForPhone = async <S extends WalletCurrency>({
+  phone,
+  balanceAmount,
+}: {
+  phone: PhoneNumber
+  balanceAmount: PaymentAmount<S>
+}) => {
+  // Create new wallet
+  const wallet = await createNewWalletFromPhone({
+    phone,
+    currency: balanceAmount.currency,
+  })
+  if (wallet instanceof Error) throw wallet
+
+  // Fund new wallet if a non-zero balance is passed
+  if (balanceAmount.amount === 0n) return wallet
+
+  const lnInvoice = await Wallets.addInvoiceForSelf({
+    walletId: wallet.id,
+    amount: Number(balanceAmount.amount),
+    memo: `Fund new wallet ${wallet.id}`,
+  })
+  if (lnInvoice instanceof Error) throw lnInvoice
+  const { paymentRequest: invoice, paymentHash } = lnInvoice
+
+  const updateInvoice = () =>
+    Wallets.updatePendingInvoiceByPaymentHash({
+      paymentHash,
+      logger: baseLogger,
+    })
+
+  const promises = Promise.all([
+    safePay({ lnd: lndOutside1, request: invoice }),
+    (async () => {
+      // TODO: we could use event instead of a sleep to lower test latency
+      await sleep(500)
+      return updateInvoice()
```

Comment:
I just tried this and the test timed out because I assume it doesn't get past the `await safePay` line to execute the update.

I also tried calling `safePay` without the `await` and the test still timed out.